src/contracts/BConst.sol

```diff
@@ -41,7 +41,7 @@ contract BConst {
   uint256 public constant BPOW_PRECISION = BONE / 10 ** 10;
 
   /// @notice The maximum ratio of input tokens vs the current pool balance.
-  uint256 public constant MAX_IN_RATIO = BONE / 2;
+  uint256 public constant MAX_IN_RATIO = BONE >> 1;
```

Comment:
I checked and this _does_ save some gas, and have some questions:

- why? it should be known at compile time
- is it related to bitshift not being checked arithmetic? shouldn't that be optimized away by the compiler? (given it can know a constant of 2 will never be zero)
- shouldn't this *always* be optimized by the compiler?

Reply:
the gas savings are minimal from what i've seen, but answering questions:
- surprisingly not, calculated constants don't get evaluated and stored, but their calc gets stored in the contract and called whenever the constant is instantiated (try `uint256 constant SHOULD_FAIL = uint256(0) - 1`)
- bitshift doesn't check arithmetic
- i think it SHOULD be optimized by the compiler, tho it does not